subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/conflicts/DefaultConflictResolutionResult.java

```diff
@@ -16,19 +16,25 @@
 
 package org.gradle.api.internal.artifacts.ivyservice.resolveengine.graph.conflicts;
 
+import org.gradle.api.Action;
+import org.gradle.api.artifacts.ModuleIdentifier;
 import org.gradle.api.internal.artifacts.ivyservice.resolveengine.ComponentResolutionState;
 
+import java.util.Collection;
+
 class DefaultConflictResolutionResult implements ConflictResolutionResult {
-    private final PotentialConflict conflict;
+    private final Collection<? extends ModuleIdentifier> participatingModules;
```

Comment:
❌  Why are we inlining the `PotentialConflict` here? That seems like a too eager optimization.

Reply:
Because it's no longer `Potential` at all. It's a real conflict, with real modules. Look at the last commit comment for more info. I think it's a good move to niline.

Reply:
I see, that makes sense.